lib/ansible/modules/files/copy.py

```diff
@@ -269,14 +269,33 @@
 import traceback
 
 from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.common.process import get_bin_path
 from ansible.module_utils._text import to_bytes, to_native
 
 
+# The AnsibleModule object
+module = None
+
+
 class AnsibleModuleError(Exception):
     def __init__(self, results):
         self.results = results
 
 
+# Once we get run_command moved into common, we can move this into a common/files module.  We can't
+# until then because of the module.run_command() method.  We may need to move it into
+# basic::AnsibleModule() until then but if so, make it a private function so that we don't have to
+# keep it for backwards compatibility later.
+def clear_facls(path):
```

Comment:
this is too aggressive as it removes all ACLs, not just the ones we added

take into account that directories can have 'default acl' to be added to all new files in a directory, this would wipe those along with the one we added to copy the file into place

Reply:
There are no default ACLs at that point as they are overridden by `shutil.copystat`. Which leaves us with irrelevant ACLs so clearing all of them makes sense.

See explaining comments including the FIXME note about default ACLs I added in the last commit. I am very open to suggestions as I don't have better ideas.

Reply:
hmm, need to x2 check but it used to keep default ACLs ... or is this a PY2 vs PY3 issue?